### ISSUE TITLE
Docker on Monorepo

### ISSUE THREAD

**Issue by yovanoc** [prisma/prisma2#400]
Hello, I'm working on a monorepo with yarn workspaces and I would like to use Docker for alternate with dev and prod mode and it feels a little bit complicated.

![image](https://user-images.githubusercontent.com/7302317/63581440-577c2c80-c597-11e9-9df4-8ce3aed555cf.png)

This is the architecture of the project. And I was telling me that it would be great to have an example to make this working because a lot of people could be interested.

Thanks

**Comment by yovanoc** [comment]
I would like to add if we can setup a postgres or other database system with docker-compose too, could be nice, thanks

**Comment by divyenduz** [comment]
There is a docker example now: https://github.com/prisma/prisma-examples/tree/prisma2/deployment-platforms/docker

Not specifically with `docker-compose` + `yarn` workspaces though but if this already solves your issue, please feel free to close it.

**Comment by yovanoc** [comment]
No I’m really struggling with the monorepo things and docker. Like how to take only packages dist folder and put these together with the prisma folder and run all of these things. I’m new with docker and I was thinking that an example of that with docker compose and a yarn workspaces would be great because it’s harder than a simple docker example like this one that I understand and I think that other people could be interested too

**Comment by yovanoc** [comment]
As @schickling pointed https://github.com/prisma/prisma2/issues/54#issuecomment-538898490 here's my issue about a more specific doc/tutorial. I still think it would be good to have a monorepo example with yarn workspaces and kubernetes maybe.

**Comment by yovanoc** [comment]
I tried today to experiment this and I'm quite confused again. I see the PR for the new @prisma/photon way and we have to install prisma2 as a dev dependency but with docker multi stage builds we can't use prisma2 generate in the final stage because it's only for production things

**Comment by pantharshit00** [comment]
I am not sure about what you mean by this. We are still generating the photon api in the `node_module`. Facade was done so that we can get rid of issues caused by not having photon in package.json. Also, you will need to add `@prisma/photon` in the package.json instead of `prisma2`. CLI is still optional.

**Comment by yovanoc** [comment]
I'm trying to use this https://docs.docker.com/develop/develop-images/multistage-build/
In my last stage I only build node_modules used for production, so only `dependencies` and not `devDependencies` and I get this error
![image](https://user-images.githubusercontent.com/7302317/69544156-6eb02900-0f8f-11ea-97a2-836ea44ee499.png)

We have to `prisma2 generate` but we can't because prisma2 have to be in `devDependencies` and only `dependencies` are available here.

I don't know if I'm well explaining this

**Comment by pantharshit00** [comment]
You mean the generated API is not persisted because you prune the node_modules. In that case, you can choose to generate photon somewhere else or install the cli globally and run it after you install the devDependencies.

**Comment by Sytten** [comment]
I created a guide that might help you https://medium.com/@emilefugulin/building-a-sane-docker-image-for-typescript-lerna-and-prisma-2-76d8ff9926e4

**Comment by yovanoc** [comment]
oh great explanation, but i'm kinda noob in docker things and I try to learn with docker-compose right now for multiple packages in the workspace, then I really don't know how to have a dev and a prod environment with that kind of setup..

**Comment by nnennajohn** [comment]
@Sytten would you happen to have a guide on the same setup but for local development, like where it compiles continuously and syncs files to docker? Thanks.

**Comment by yovanoc** [comment]
Yes thats what I said by "dev and prod environment", and docker-compose/kubernetes would be the cherry on top 😂

**Comment by Sytten** [comment]
Not really, but it should be fairly easy to do if you mount the directory directly. Also my guide is not up to date with the latest beta.4 changes.
